File: AndroidStudio/NewBuilding/app/src/main/java/com/example/newbuilding/MainActivity.java
package com.example.newbuilding;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.example.newbuilding.databinding.ActivityMainBinding;
import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity {

    /*
    // Comoonentes
    private EditText etNombre;
    private EditText enEdad;
    private EditText epContrasena;
    private Button btGuardar;
     */
    Usuario usuario;
    ActivityMainBinding binding;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        //recuperarComponentesInterfaz();
        asignarEventos();

        // setContentView(R.layout.activity_main);
        setContentView(binding.getRoot());

    }

    /*
    //Metodo para recuperar componentes de la interfaz
    private void recuperarComponentesInterfaz(){
        etNombre = (EditText) findViewById(R.id.id_Nombre_EditText);
        enEdad = (EditText) findViewById(R.id.id_Edad_EditNumber);
        epContrasena = (EditText) findViewById(R.id.id_Contrasena_EditPassword);
        btGuardar = (Button) findViewById(R.id.id_Guardar_Button);

    }
    */

    private void asignarEventos() {
        binding.idGuardarButton.setOnClickListener(view -> {
            String nombre = binding.idNombreEditText.getText().toString();
            int edad = Integer.getInteger(binding.idEdadEditNumber.getText().toString());
            String contrasena = binding.idContrasenaEditPassword.getText().toString();

            binding.setPersona(new Usuario(nombre,edad,contrasena));

            /*
            asignarDatos(new Usuario(nombre, edad, contrasena));

            String nombre = etNombre.getText().toString();
            int edad = Integer.getInteger(enEdad.getText().toString());
            String contrasena = epContrasena.getText().toString();
            */
            Snackbar.make(binding.getRoot(), "Datos guardados", Snackbar.LENGTH_SHORT).show();
        });
    }

/*    private void asignarDatos(Usuario usuario) {
        binding.idNombreEditText.setText(usuario.getNombre());
        binding.idEdadEditNumber.setText(usuario.getEdad());
        binding.idContrasenaEditPassword.setText(usuario.getContrsena());
    }*/

}

File: Moviles/WorkSpace Moviles/Ejem21_ToolBar/app/src/main/java/com/example/ejem21_toolbar/MainActivity.kt
package com.example.ejem21_toolbar

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import com.example.ejem21_toolbar.databinding.ActivityMainBinding
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity() {
    lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.myToolBar?.miToolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(false) // show the back button
        supportActionBar?.title = "App bar"
        supportActionBar?.subtitle = "Ejemplo de App bar"

        binding.btPass.setOnClickListener {
            startActivity(Intent(this, MainActivity2::class.java))
        }
    }
}
